packages/hdwallet-portis/src/bitcoin.ts

```diff
@@ -0,0 +1,280 @@
+      scriptType,
+      isPrefork
+    }
+  }
+}
+
+
+export async function btcGetAddress (msg: BTCGetAddress, portis: any): Promise<string> {
+
+  if(!msg.addressNList.length)
+    throw new Error('Empty addressNList')
+
+  const scriptType = msg.scriptType
+  const purpose = msg.addressNList[0]
+
+  const hardPath = hardenedPath(msg.addressNList)
+  const hardPathString = 'm/'+hardPath.reduce( (acc, val, i) => (i !== hardPath.length - 1) ? acc + (val - 2147483648) + `'/` : acc + (val - 2147483648) + `'`, '')
+
+  const { result: xpub } = await portis.getExtendedPublicKey(hardPathString, "Bitcoin")
+
+  const relPath = relativePath(msg.addressNList)
+  const relPathString = relPath.reduce( (acc, val, i) => (i !== relPath.length - 1) ? acc + val + '/' : acc + val, '')
+
+  const args = { pubkey: fromBase58(xpub).derivePath(relPathString).publicKey }
+
+  let result
+  switch (scriptType) {
+    case BTCInputScriptType.SpendAddress:
+      result = payments.p2pkh(args)
+      break
+    case BTCInputScriptType.SpendWitness:
+      result = payments.p2wpkh(args)
+      break
+    case BTCInputScriptType.SpendP2SHWitness:
+      result = payments.p2sh({
+        redeem: payments.p2wpkh(args)
+      })
+      break
+    default:
+      throw new Error(`Unsupported scriptType ${scriptType}`)
+  }
+
+  if(msg.showDisplay === true) {
+    if (!verifyScriptTypePurpose(scriptType, purpose)) {
+      throw new Error(`Invalid scriptType ${scriptType} for purpose ${purpose}`)
+    }
+
+    portis.showBitcoinWallet(addressNListToBIP32(msg.addressNList))
+  }
+
+  return result.address
+}
+
+export function verifyScriptTypePurpose(scriptType: BTCInputScriptType, purpose: number): boolean {
+  return (
+      (purpose === 0x80000000 + 44 && scriptType === BTCInputScriptType.SpendAddress ) ||
+      (purpose === 0x80000000 + 49 && scriptType === BTCInputScriptType.SpendP2SHWitness ) ||
+      (purpose === 0x80000000 + 84 && scriptType === BTCInputScriptType.SpendWitness )
+  )
+}
+
+export function legacyAccount (coin: Coin, slip44: number, accountIdx: number): BTCAccountPath {
+  return {
+    coin,
+    scriptType: BTCInputScriptType.SpendAddress,
+    addressNList: [ 0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + accountIdx ]
+  }
+}
+
+export function segwitAccount (coin: Coin, slip44: number, accountIdx: number): BTCAccountPath {
+  return {
+    coin,
+    scriptType: BTCInputScriptType.SpendP2SHWitness,
+    addressNList: [ 0x80000000 + 49, 0x80000000 + slip44, 0x80000000 + accountIdx ]
+  }
+}
+
+export function segwitNativeAccount (coin: Coin, slip44: number, accountIdx: number): BTCAccountPath {
+  return {
+    coin,
+    scriptType: BTCInputScriptType.SpendWitness,
+    addressNList: [ 0x80000000 + 84, 0x80000000 + slip44, 0x80000000 + accountIdx ]
+  }
+}
+
+
+export function btcNextAccountPath (msg: BTCAccountPath): BTCAccountPath | undefined {
+  let description = describeUTXOPath(msg.addressNList, msg.coin, msg.scriptType)
```

Comment:
Does portis support more than one BTC account? If not, this whole function should be `return undefined`

Reply:
good call.  i thought i had this returning undefined already...